app.py

```diff
@@ -10,23 +10,50 @@
 #from flask import Flask, render_template, request, jsonify, Blueprint
 
 import web
+from web import form
 import signal
 from json import dumps
+import pymongo
 from pymongo import MongoClient
 from uuid import uuid4
 from datetime import datetime
 from bson import json_util
 from threading import Condition
 import httpagentparser
 from os import _exit
-
 import time
+import math
+
+
 import config
 
+# hashing
+import hmac
```

Comment:
as before: please try to import the minimal set, use `from ... import ...` and then use the code subsequently without the module prefix where possible.